### PR TITLE
XRDDEV-1183 Client flow improvement

### PR DESCRIPTION
**Pull request by Riippi**
https://jira.niis.org/browse/XRDDEV-1183

- Add client flow handles existing csr / certificate scenarios in similar way as add member flow
- Some cleanup for add member flow

Merge PR 606 before this
https://github.com/nordic-institute/X-Road/pull/606

### CODE REVIEW

File: src/proxy-ui-api/frontend/src/views/AddClient/FinishPage.vue [hunk 1]
```diff
@@ -115,21 +163,66 @@ export default Vue.extend({
...
+        .finally(() => {
           this.disableCancel = false;
           this.submitLoading = false;
-        },
+        });
+    },
+
+    registerClient(): void {
+      const clientId = createClientId(
+        this.currentSecurityServer.instanceId,
```
Comment (raits) [inline src/proxy-ui-api/frontend/src/views/AddClient/FinishPage.vue:206]: This doesn't seem to work, API call that results from this is "/api/v1/clients/undefined:ORG:TEST/register" (instance is undefined), resulting in failure to register the client.

I think this identifier should also include the subsystem that is being added.

File: src/proxy-ui-api/frontend/src/views/AddClient/ClientDetailsPage.vue [hunk 2]
```diff
@@ -180,9 +185,9 @@ export default Vue.extend({
   data() {
     return {
       disableDone: false,
-      certificationService: undefined,
       filteredServiceList: [],
```
Comment (raits) [inline src/proxy-ui-api/frontend/src/views/AddClient/ClientDetailsPage.vue:188]: This doesn't seem to be used anymore, maybe it should be removed?

File: src/proxy-ui-api/frontend/src/views/AddClient/FinishPage.vue (hunk 1 above)
Comment (Riippi) [reply src/proxy-ui-api/frontend/src/views/AddClient/FinishPage.vue]: Fixed

File: src/proxy-ui-api/frontend/src/views/AddClient/ClientDetailsPage.vue (hunk 2 above)
Comment (Riippi) [reply src/proxy-ui-api/frontend/src/views/AddClient/ClientDetailsPage.vue]: Removed it and also another unused value

**Comment by raits** [comment]
Looks good to me.

**Pull request merged**